api/src/oscc.c

```diff
@@ -1092,3 +1092,84 @@ oscc_result_t get_device_name( char * string, char * const name )
 
     return result;
 }
+
+oscc_result_t get_wheel_speed_right_rear(
+    struct can_frame const * const frame,
+    double * wheel_speed_right_rear)
+{
+    if((frame == NULL) || (wheel_speed_right_rear == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[7] & 0x0F) << 8) | frame->data[6];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_right_rear = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_wheel_speed_left_rear(
```

Comment:
It seems like the conversion logic is common for all of the wheels and the only difference is which frames are used. This could be converted into a single function loosely resembling this:

```coscc_result_t get_wheel_speed_x(
    struct can_frame const * const frame,
    double * wheel_speed,
    uint8_t offset)
{
    if((frame == NULL) || (wheel_speed == NULL)) { return OSCC_ERROR; }
    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
    uint16_t raw = ((frame->data[offset + 1] & 0x0F) << 8) | frame->data[offset];
    // 10^-1 precision, raw / 32.0
    *wheel_speed = (double)((int)((double)raw / 3.2) / 10.0);
    return OSCC_OK;
}
```

I'll leave it up to others to decide if this is a blocking change.

Reply:
Oh yeah, upon re-reading my first comment I realize that I phrased things poorly. I much prefer the API with the 4 functions and just want to have them effectively act as closures that bind the values of the offset into the common function.

Anywho, I'll stop bike shedding. :)

Reply:
Ah! I get it, I'm on board.

Reply:
Ah okay I get it, good call!